web/src/beta/lib/reearth-ui/components/PopupMenu/index.tsx

```diff
@@ -67,34 +69,65 @@ export const PopupMenu: FC<PopupMenuProps> = ({
     [onOpenChange, open],
   );
 
+  const renderSingleItem = (item: PopupMenuItem, index: number) => {
+    const { icon, id, onClick, path, selected, subItem, title } = item;
+    return (
+      <Item
+        key={index}
+        size={size}
+        onClick={() => {
+          onClick?.(id);
+          handlePopOver(false);
+        }}>
+        {icon && (
+          <Icon icon={icon} size="small" color={iconColor ? iconColor : theme.content.weak} />
+        )}
+        <div
+          style={{
+            display: "flex",
+            justifyContent: "space-between",
+            justifyItems: "center",
+            flexGrow: 1,
+          }}>
+          {subItem ? (
+            <PopupMenu label={title} menu={subItem} width={width} nested />
+          ) : path ? (
+            <StyledLink to={path}>
+              <Typography size="body">{title}</Typography>
+            </StyledLink>
+          ) : (
+            <Typography size="body">{title}</Typography>
+          )}
+          {selected && <Icon icon="check" size="small" color={theme.content.main} />}
+        </div>
+      </Item>
+    );
+  };
+
+  const renderSubMenuItems = (subMenuItems: PopupMenuItem[]) => {
+    const personalWorkspaces = subMenuItems.filter(item => !!item.personal);
```

Comment:
Since this component is re-usable may be could improve naming, this looks like it only for workspace???

Reply:
Yes, but I think `renderSubMenuItems` is custom and can only be rendered if a custom property is defined.

Reply:
this one should be more generic but now it to customized for the workspace i.e using team and personal direct as title in here is not sound good what if there is other component with the need to use this logic of grouping menus????

Reply:
Thanks @mkumbobeaty. I've worked on this to make it more reusable.

Reply:
Update confirmed. We can refactor more in the future if needed.